src/amo/components/Header/index.js

```diff
@@ -67,6 +67,9 @@ export class HeaderBase extends React.Component {
     const viewProfileURL = siteUser ? `/user/${siteUser.username}/` : null;
     const viewProfileLinkProps = config.get('enableUserProfile') ?
       { to: viewProfileURL } : { href: viewProfileURL };
+    const editProfileURL = siteUser ? '/users/edit' : null;
+    const editProfileLinkProps = config.get('enableUserProfile') ?
+      { to: editProfileURL } : { href: editProfileURL };
```

Comment:
If I remember correctly, `to` is intercepted by the react-router and always to navigate without page reload. `href` is used for plain old HTML links, which we have to use since the edit/view pages are currently served by the old frontend.